analyzer/protocol/ldap/ldap.spicy

```diff
@@ -0,0 +1,638 @@
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Modify DN Operation
+# https://tools.ietf.org/html/rfc4511#section-4.8
+
+type ModDNRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse mod DN request sequence
+
+  on %done {
+    print "ModDNRequest", message.messageID, self;
+  }
+};
+
+type ModDNResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "ModDNResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Compare Operation
+# https://tools.ietf.org/html/rfc4511#section-4.10
+
+type CompareRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse compare request entry/ava sequence
+
+  on %done {
+    print "CompareRequest", message.messageID, self;
+  }
+};
+
+type CompareResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "CompareResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Abandon Operation
+# https://tools.ietf.org/html/rfc4511#section-4.11
+
+type AbandonRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+  # messageID: asn1::ASN1Message(True, self.depth) &convert=$$.body.num_value;
+
+  on %done {
+    # TODO: can't do string -> bytes
+    # but see https://github.com/zeek/spicy/issues/911 for next release
+    # message.obj = "%d" % (self.messageID);
+    print "AbandonRequest", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Extended Operation
+# https://tools.ietf.org/html/rfc4511#section-4.12
+
+type ExtendedRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse ExtendedRequest sequence
+
+  on %done {
+    print "ExtendedRequest", message.messageID, self;
+  }
+};
+
+type ExtendedResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse ExtendedRequest sequence
+
+  on %done {
+    # message.result = self.result;
+    print "ExtendedResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# IntermediateResponse Message
+# https://tools.ietf.org/html/rfc4511#section-4.13
+
+type IntermediateResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse IntermediateResponse sequence
```

Comment:
Is it correct that this parser will consume no data? If not, let's instead either skip over its data, or leave out of `Message` for now.